pulp_ansible/app/viewsets.py

```diff
@@ -88,6 +89,47 @@ class RoleViewSet(ContentViewSet):
     queryset = Role.objects.all()
     serializer_class = RoleSerializer
     filterset_class = RoleFilter
+    queryset_filtering_required_permission = "ansible.view_role"
+
+    DEFAULT_ACCESS_POLICY = {
+        "statements": [
+            {
+                "action": ["list", "retrieve"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_repository_model_or_obj_perms:ansible.view_role",
+            },
+            {
+                "action": ["create"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": [
+                    "has_required_repo_perms_on_upload:ansible.modify_ansible_repo_content",
+                    "has_required_repo_perms_on_upload:ansible.view_ansiblerepository",
+                    "has_upload_param_model_or_obj_perms:core.change_upload",
+                ],
+            },
+            {
+                "action": ["destroy"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": [
+                    "has_repository_model_or_obj_perms:ansible.delete_role",
+                    "has_repository_model_or_obj_perms:ansible.view_role",
+                ],
+            },
+            {
+                "action": ["update"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": [
+                    "has_repository_model_or_obj_perms:ansible.change_role",
+                    "has_repository_model_or_obj_perms:ansible.view_role",
+                ],
+            },
```

Comment:
There are no destroy or update actions for Content in Pulp APIs, so I would remove these.